internal/npm_install/generate_build_file.js

```diff
@@ -155,6 +158,41 @@ function isPackage(p) {
       fs.statSync(packageJson).isFile();
 }
 
+/**
+ * Returns true if directory is a Bazel package directory.
+ */
+function isPackageDir(d) {
+  return fs.existsSync(path.posix.join(d, 'BUILD')) ||
+      fs.existsSync(path.posix.join(d, 'BUILD.bazel'));
+}
+
+/**
+ * Returns an array of all the files under a directory as relative
+ * paths to the directory.
+ */
+function listFiles(rootDir, subDir = '') {
+  const dir = path.posix.join(rootDir, subDir);
+  return fs
+      .readdirSync(dir)
+      // Filter out .bin folder since this is a folder created by yarn/npm that contains
+      // only symlinks and it breaks this function on Windows.
+      // For example, `node_modules/foo/.bin/foo` would be a symlink to `node_modules/.bin/foo`
+      .filter(f => !(f === '.bin' && fs.statSync(path.posix.join(dir, f)).isDirectory()))
+      // Also filter out test folders which contain symlinks in some packages
+      // such as `ecstatic/test/public/containsSymlink/more-problematic`
+      .filter(f => !(f === 'test' && fs.statSync(path.posix.join(dir, f)).isDirectory()))
+      // Filter out directories that are bazel packages so that files do not cross Bazel package
+      // boundaries. npm packages should not generally include BUILD or BUILD.bazel files
+      // but they may as rxjs does temporarily.
+      .reduce((files, file) => {
+        const fullPath = path.posix.join(dir, file);
+        const relPath = path.posix.join(subDir, file);
+        return fs.statSync(fullPath).isDirectory() ?
+            (isPackageDir(fullPath) ? files : files.concat(listFiles(rootDir, relPath))) :
```

Comment:
rules_node just deletes the BUILD/BUILD.bazel files instead
https://github.com/pubref/rules_node/blob/master/node/internal/clean_node_modules.js#L9
what do you think are the tradeoffs? I can imagine a package is published with a BUILD file in the root because they use bazel and forgot to .npmignore it, and now we filter out all fines in that package?

Reply:
Deleting the BUILD files sounds reasonable. I think I prefer that.